### PR TITLE
There is no need to run gather_odf_client script on non-provider setup

### PR DESCRIPTION
**Pull request by OdedViner**

### CODE REVIEW

**Comment by yati1998** [comment]
@OdedViner did you verify your image in both provider and non-provider clusters?

**Comment by OdedViner** [comment]
> @OdedViner did you verify your image in both provider and non-provider clusters?
```
podman build -t quay.io/oviner/ocs-must-gather:provider_mode_1 -f Dockerfile .
podman push quay.io/oviner/ocs-must-gather:provider_mode_1


Running on provider cluster:
$ oc adm must-gather --image=quay.io/oviner/ocs-must-gather:provider_mode_1 > provider_cluster.log
$ cat provider_cluster.log | grep "it is provider"
[must-gather-zvrqm] POD 2024-09-19T08:44:47.880752513Z it is provider cluster


Running on "regular cluster":
$ oc adm must-gather --image=quay.io/oviner/ocs-must-gather:provider_mode_1 > non_provider_cluster.log
$ cat non_provider_cluster.log | grep "it is not"
[must-gather-xj2dw] POD 2024-09-19T09:02:59.573594086Z it is not provider cluster
```

**Review by yati1998 — APPROVED** [review]

**Comment by black-dragon74** [comment]
What if the current cluster is a consumer? Is that accounted for as well?

**Comment by yati1998** [comment]
> What if the current cluster is a consumer? Is that accounted for as well?

if I am not wrong yes, by provider he means: provider-consumer cluster.
cc @OdedViner

File: collection-scripts/gather_main [hunk 1]
```diff
@@ -21,8 +21,15 @@ procids+=($!)
 gather_dr_resources &
 procids+=($!)
 
-gather_odf_client &
-procids+=($!)
+storagecluster_remote=$(oc get storagecluster --all-namespaces  -o yaml | grep -i 'allowRemoteStorageConsumers')
```
Comment (jilju) [inline collection-scripts/gather_main:24]: Storagecluster CRD won't be present in client cluster on a provider-client(provider mode) setup.

Comment (yati1998) [reply collection-scripts/gather_main:24]: @OdedViner please re-verify the check and paste the logs collected.
According to the check you mentioned, odf_client script will not run even if it is a client cluster.

**Comment by black-dragon74** [comment]
Please do all the custom checks in sub gather scripts. The `gather` script is an entrypoint and should not contain extraneous checks. I would move it to `gather_odf_client`

Also, do we need this change @yati1998 ?

**Comment by yati1998** [comment]
> Please do all the custom checks in sub gather scripts. The `gather` script is an entrypoint and should not contain extraneous checks. I would move it to `gather_odf_client`
> 
> Also, do we need this change @yati1998 ?

No we dont,
Once @OdedViner verify his again, we can close this PR

**Comment by black-dragon74** [comment]
Closing this as it is not needed.

**Pull request closed without merging**